pkg/installmanager/awsactuator.go

```diff
@@ -0,0 +1,113 @@
+package installmanager
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+	"github.com/openshift/hive/pkg/constants"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/pkg/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+func init() {
+	registerActuator(&awsActuator{awsClientFn: getAWSClient})
+}
+
+// Ensure AWSActuator implements the Actuator interface. This will fail at compile time when false.
+var _ Actuator = &awsActuator{}
+
+// AWSActuator manages getting the desired state, getting the current state and reconciling the two.
+type awsActuator struct {
+	// awsClientFn is the function to build an AWS client, here for testing
+	awsClientFn func(client.Client, string, string, string, log.FieldLogger) (awsclient.Client, error)
+}
```

Comment:
hmm.. i don't really see the benefit of this,

- globally registered list of actuators that use init function to configure them. Can we just explicitly have a list locally where it matters. Is there a benefit to this dependency injection here?
- why not just have a awsclient.Client interface directly embed in the struct and use a newAWSActuator function to create that when we know we can configure it. This style of dependency injection imo is only useful that it abstracts an external API/function not what is required right here like the client to AWS.